index.js

```diff
@@ -119,14 +111,46 @@
   //  $$type :: String
   var $$type = '@@type';
 
-  //  type :: Any -> String
+  //  RPARSE :: RegExp
+  var RPARSE = /^(?:([^/\n\r]+)\/)?(.+?)(?:@(\d+))?$/;
+
+  //       TypeIdentifier :: (a, b, c) -> TypeIdentifier a b c
```

Comment:
> We very much want to ensure that the values we return are members of a concrete type

Ah. It seems like you view these hindley-milner comments like type restrictions, where I viewed it more like documentation of the actual types.

> We should make the `version` type `Number` rather than `Nullable String` or `Nullable Number`, and default to `0` if no version is specified.

I considered this option too, but thought it'd be weird to say that every native type has version `0`.
Thinking about it again I'm inclined to prefer the consistency of the Type though, so I'll change it.